src/components/utils/src/threads/thread_posix.cc

```diff
@@ -261,15 +270,27 @@ void Thread::stop() {
   SDL_DEBUG("Stopped thread #" << handle_ << " \"" << name_ << " \"");
 }
 
-void Thread::join() {
+void Thread::join(const bool force_stop) {
   SDL_AUTO_TRACE();
   DCHECK_OR_RETURN_VOID(!IsCurrentThread());
 
-  stop();
+  if (force_stop) {
+    stop();
+  }
 
```

Comment:
@orhan-mehmedov for example you can create some kind of unprotected private method `StopUnsafe()` without mutex locking inside and call it here under lock.